v2/emailpassword/common-customizations/sessions/user-impersonation.mdx

```diff
@@ -383,12 +383,29 @@ export function POST(req: NextRequest) {
             throw new Error("User does not exist");
         }
 
-        const session = await createNewSession(baseRequest, baseResponse, "public", users[0].loginMethods[0].recipeUserId, {
-            isImpersonation: true,
-        });
+        try {
+            const session = await createNewSession(baseRequest, baseResponse, "public", users[0].loginMethods[0].recipeUserId, {
+                isImpersonation: true,
+            });
 
-        return NextResponse.json({ message: "Impersonation successful" });
+            return NextResponse.json({ message: "Impersonation successful" });
+        } catch (err) {
+            try {
+                await errorHandler()(err, baseRequest, baseResponse, (errorHandlerError: Error) => {
+                    if (errorHandlerError) throw err;
+                });
+                return new NextResponse(baseResponse.body, {
+                    status: baseResponse.statusCode,
+                    headers: baseResponse.headers,
+                });
+            } catch (error) {
+                // Handle the error
+                return NextResponse.json({ message: "Somethign went wrong!"}, { status: 500 });
+            }
+        }
```

Comment:
this should go inside withPreParsedRequestResponse

Reply:
done